mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart

```diff
@@ -311,4 +329,11 @@ class RenderList {
     GroupAssetsBy groupBy,
   ) =>
       _buildRenderList(assets, null, groupBy);
+
+  /// Deletes an asset from the render list and clears the buffer
+  /// This is only a workaround for deleted images still appearing in the gallery
+  Future<void> deleteAsset(Asset deleteAsset) async{
+    allAssets?.remove(deleteAsset);
+    _buf.clear();
+  }
```

Comment:
@fyfrey Does this look OK to you?

Reply:
To be sure: also set bufOffset=0.

Does the method need to be async at all?

Reply:
Nope, it doesn't. I also set bufOffset.